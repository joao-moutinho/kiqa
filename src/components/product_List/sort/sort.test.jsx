import {fireEvent, render, screen} from '@testing-library/react';
import { Sort } from '.';
import {BrowserRouter} from 'react-router-dom';
import { renderHook, act } from '@testing-library/react-hooks'



describe("Sort", () =>{

    test('render ' ,() => {
        const fn = jest.fn();
        const { container } = render( <BrowserRouter> <Sort setMyState={fn} searchParamsChange={fn}/> </BrowserRouter> );
        expect(container).toMatchSnapshot();
    })
  
     test('test chevron when was pressed', () => {
        const fn = jest.fn();
        const fn2 = jest.fn();
        render( <BrowserRouter> <Sort setMyState={fn} searchParamsChange={fn2}/> </BrowserRouter> );
        // eslint-disable-next-line jest/valid-expect
        expect(fireEvent.click(screen.getByTestId("chevron")))          
    }) 

    test('test dropdown Default', () => {
        const fn = jest.fn();
        const fn2 = jest.fn();
        const {container} =render( <BrowserRouter> <Sort setMyState={fn} searchParamsChange={fn2}/> </BrowserRouter> );
        // eslint-disable-next-line jest/valid-expect
        fireEvent.click(screen.getByTestId("chevron")); 
        // eslint-disable-next-line testing-library/await-async-query
        expect(screen.getByTestId("right-dropdown")).toBeInTheDocument();
        fireEvent.click(screen.getByTestId("buttonDefault"));
        const items =  screen.getAllByText(/Default/i);
        expect(items).toHaveLength(1);
        expect(container).toMatchSnapshot();
    })

    test('test dropdown 1-9 Asc', () => {
        const fn = jest.fn();
        const fn2 = jest.fn();
        const {container} =render( <BrowserRouter> <Sort setMyState={fn} searchParamsChange={fn2}/> </BrowserRouter> );
        // eslint-disable-next-line jest/valid-expect
        fireEvent.click(screen.getByTestId("chevron")); 
        // eslint-disable-next-line testing-library/await-async-query
        expect(screen.getByTestId("right-dropdown")).toBeInTheDocument();
        // eslint-disable-next-line jest/valid-expect
        const buttonAsc = screen.getByText(/1-9 Asc/i);
        fireEvent.click(buttonAsc);
        const items =  screen.getAllByText(/1-9 Asc/i);
        expect(items).toHaveLength(1);
        expect(container).toMatchSnapshot();
       
    }) 

    test('test dropdown 1-9 Des', () => {
        const fn = jest.fn();
        const fn2 = jest.fn();
        const {container} =render( <BrowserRouter> <Sort setMyState={fn} searchParamsChange={fn2}/> </BrowserRouter> );
        // eslint-disable-next-line jest/valid-expect
        fireEvent.click(screen.getByTestId("chevron")); 
        // eslint-disable-next-line testing-library/await-async-query
        expect(screen.getByTestId("right-dropdown")).toBeInTheDocument();
        // eslint-disable-next-line jest/valid-expect
        const buttonDesc = screen.getByText(/9-1 Desc/i);
        fireEvent.click(buttonDesc);
        const items =  screen.getAllByText(/9-1 Desc/i);
        expect(items).toHaveLength(1);
        expect(container).toMatchSnapshot();
    }) 

    test('test dropdown A-Z Asc', () => {
        const fn = jest.fn();
        const fn2 = jest.fn();
        const {container} =render( <BrowserRouter> <Sort setMyState={fn} searchParamsChange={fn2}/> </BrowserRouter> );
        // eslint-disable-next-line jest/valid-expect
        fireEvent.click(screen.getByTestId("chevron")); 
        // eslint-disable-next-line testing-library/await-async-query
        expect(screen.getByTestId("right-dropdown")).toBeInTheDocument();
        // eslint-disable-next-line jest/valid-expect
        const buttonDesc = screen.getByText(/A-Z Asc/i);
        fireEvent.click(buttonDesc);
        const items =  screen.getAllByText(/A-Z Asc/i);
        expect(items).toHaveLength(1);
        expect(container).toMatchSnapshot();
    }) 

    test('test dropdown Z-A Desc', () => {
        const fn = jest.fn();
        const fn2 = jest.fn();
        const {container} =render( <BrowserRouter> <Sort setMyState={fn} searchParamsChange={fn2}/> </BrowserRouter> );
        // eslint-disable-next-line jest/valid-expect
        fireEvent.click(screen.getByTestId("chevron")); 
        // eslint-disable-next-line testing-library/await-async-query
        expect(screen.getByTestId("right-dropdown")).toBeInTheDocument();
        // eslint-disable-next-line jest/valid-expect
        const buttonDesc = screen.getByText(/Z-A Desc/i);
        fireEvent.click(buttonDesc);
        const items =  screen.getAllByText(/Z-A Desc/i);
        expect(items).toHaveLength(1);
        expect(container).toMatchSnapshot();
    }) 
})