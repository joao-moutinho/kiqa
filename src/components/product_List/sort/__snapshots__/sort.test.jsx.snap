// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort render  1`] = `
<div>
   
  <div
    class="sort"
  >
    <div
      class="sort-box"
    >
      <div
        class="sort-box-left"
      >
        <nav
          arial-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-hn8qoe-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/products"
              >
                Products
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="sort-box-right"
      >
        <div
          class="sort-box-right-title"
        >
           Sort by: 
        </div>
        <div
          class="sort-box-right-output"
          data-testid="output"
        >
          Default
          <svg
            class="sort-box-right-output-iconclose"
            data-testid="chevron"
            fill="currentColor"
            height="25"
            name="chevron"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`Sort test dropdown 1-9 Asc 1`] = `
<div>
   
  <div
    class="sort"
  >
    <div
      class="sort-box"
    >
      <div
        class="sort-box-left"
      >
        <nav
          arial-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-hn8qoe-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/products"
              >
                Products
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="sort-box-right"
      >
        <div
          class="sort-box-right-title"
        >
           Sort by: 
        </div>
        <div
          class="sort-box-right-output"
          data-testid="output"
        >
          1-9 Asc
          <svg
            class="sort-box-right-output-iconclose"
            data-testid="chevron"
            fill="currentColor"
            height="25"
            name="chevron"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`Sort test dropdown 1-9 Des 1`] = `
<div>
   
  <div
    class="sort"
  >
    <div
      class="sort-box"
    >
      <div
        class="sort-box-left"
      >
        <nav
          arial-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-hn8qoe-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/products"
              >
                Products
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="sort-box-right"
      >
        <div
          class="sort-box-right-title"
        >
           Sort by: 
        </div>
        <div
          class="sort-box-right-output"
          data-testid="output"
        >
          9-1 Desc
          <svg
            class="sort-box-right-output-iconclose"
            data-testid="chevron"
            fill="currentColor"
            height="25"
            name="chevron"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`Sort test dropdown A-Z Asc 1`] = `
<div>
   
  <div
    class="sort"
  >
    <div
      class="sort-box"
    >
      <div
        class="sort-box-left"
      >
        <nav
          arial-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-hn8qoe-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/products"
              >
                Products
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="sort-box-right"
      >
        <div
          class="sort-box-right-title"
        >
           Sort by: 
        </div>
        <div
          class="sort-box-right-output"
          data-testid="output"
        >
          A-Z Asc
          <svg
            class="sort-box-right-output-iconclose"
            data-testid="chevron"
            fill="currentColor"
            height="25"
            name="chevron"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`Sort test dropdown Default 1`] = `
<div>
   
  <div
    class="sort"
  >
    <div
      class="sort-box"
    >
      <div
        class="sort-box-left"
      >
        <nav
          arial-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-hn8qoe-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/products"
              >
                Products
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="sort-box-right"
      >
        <div
          class="sort-box-right-title"
        >
           Sort by: 
        </div>
        <div
          class="sort-box-right-output"
          data-testid="output"
        >
          Default
          <svg
            class="sort-box-right-output-iconclose"
            data-testid="chevron"
            fill="currentColor"
            height="25"
            name="chevron"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`Sort test dropdown Z-A Desc 1`] = `
<div>
   
  <div
    class="sort"
  >
    <div
      class="sort-box"
    >
      <div
        class="sort-box-left"
      >
        <nav
          arial-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-hn8qoe-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-azkret-MuiTypography-root-MuiLink-root"
                href="/products"
              >
                Products
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="sort-box-right"
      >
        <div
          class="sort-box-right-title"
        >
           Sort by: 
        </div>
        <div
          class="sort-box-right-output"
          data-testid="output"
        >
          Z-A Desc
          <svg
            class="sort-box-right-output-iconclose"
            data-testid="chevron"
            fill="currentColor"
            height="25"
            name="chevron"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
</div>
`;
